### ISSUE TITLE
Task.cc:2114:read_bytes_helper() errno: EIO

### ISSUE THREAD

**Issue by aarzilli** [mozilla/rr#2033]
I'm getting this error with binaries built with go, it's a version of go built from git but I doubt it's go that's the problem.

I see that there was a similar issue last month so I made sure to pull and build rr from source: 

```
$ git rev-parse HEAD
5f0f8b0c6155a92134ecfaa25fc1f9fff9080b26
```

The full error is:

```
$ rr record ./continuetestprog

rr: Warning: Your CPU frequency governor is 'powersave'.  rr strongly
    recommends that you use the 'performance' governor.  Not using the
    'performance' governor can cause rr to be at least 2x slower
    on laptops.

    On Fedora-based systems, you can enable the 'performance' governor
    by running the following commands:

    $ sudo dnf install kernel-tools
    $ sudo cpupower frequency-set -g performance

rr: Saving execution to trace directory `/home/a/.rr/continuetestprog-0'.
[FATAL /home/a/n/rr/src/Task.cc:2114:read_bytes_helper() errno: EIO] 
 (task 29513 (rec:29513) at time 337)
 -> Assertion `false' failed to hold. Should have read 8 bytes from 0, but only read -1
Launch gdb with
  gdb '-l' '10000' '-ex' 'target extended-remote :29513' /home/a/n/go/src/github.com/derekparker/delve/_fixtures/continuetestprog
```

Additionally if I try to replay the recording a 'continue' will never exit, and the rr process is pegged at 100% CPU.

Statically linked executable to reproduce this issue and rr trace directory available here https://github.com/aarzilli/drop/blob/master/continuetestprog.tgz

**Comment by khuey** [comment]
Thank you for the report.  This should be fixed by e76c53f0fe61d70033a1d91c31bffcd9938d29e0.

**Comment by aarzilli** [comment]
Works well, thank you.
